pstoremem/addr_book.go

```diff
@@ -56,16 +150,20 @@ var _ pstore.AddrBook = (*memoryAddrBook)(nil)
 func NewAddrBook() pstore.AddrBook {
 	ctx, cancel := context.WithCancel(context.Background())
 
+	makeSegments := func() (ret addrSegments) {
+		for i, _ := range ret {
+			ret[i] = &addrSegment{addrs: make(map[peer.ID]map[string]*expiringAddr)}
+		}
+		return ret
+	}
+
 	ab := &memoryAddrBook{
-		segments: func() (ret addrSegments) {
-			for i, _ := range ret {
-				ret[i] = &addrSegment{addrs: make(map[peer.ID]map[string]*expiringAddr)}
-			}
-			return ret
-		}(),
-		subManager: NewAddrSubManager(),
-		ctx:        ctx,
-		cancel:     cancel,
+		segments:            makeSegments(),
+		signedSegments:      makeSegments(),
```

Comment:
we probably want to aggregate those two.

Reply:
it is much easier to follow now that they're combined! 👍
I also put the `signedRoutingStates` map into the `addrSegment`, instead of having a single map inside the `memoryAddressBook` struct. I realized that I didn't have any locks around that map and was mutating it in a few places, which seems like it was bound to lead to trouble.